### ISSUE TITLE
Popover - Ensure content updates define focus updates appropriately - (2036570080)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5593]
### Module: 
 23b Popover 

### Violation: 
 Ensure content updates define focus updates appropriately 

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- [2.4.3 Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) 

### Instance ID: 
 2036570080 

### Severity: 
 8 

### Description: 
 [Issue]
Focus does not come back to "Activate Popover with Close Button". when actives "Close" button of dialog.

[User Impact]
When content within a page changes without reloading the page and without appropriate focus changes, assistive technologies such as screen readers may not be aware of the changes or the user may only be alerted that a change in content occurred and have to review the entire page to discover exactly what changed.

[Code Reference] 
 ```
<button aria-busy="false" aria-disabled="false" aria-label="Close" class="button"><div aria-hidden="true" class="icon-container"><div class="slot-container"><slot></slot></div></div><div class="text-container">Close</div></button>
 ``` 

### Note: 
 [Suggestion]
When the context of the page changes such as when the user initiates a change that updates a large portion of the page (e.g. in a single page application) focus should move to the beginning of the updated content. Similarly, when a button is activated to open a dialog, focus should be moved into the dialog, and when the dialog is closed focus needs to return to the element that triggered the dialog.

[Compliant Code Example] 
 ```
N/A
 ```
 

### Media Type: 
 ARIA and Dynamic Content 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
Also of note that the `popover` close button follows DOM order, so if listed before the reference element, will not be listed in the expected order to the user.

Current workaround for this, and the above issue related to the focus: https://codepen.io/geospatialem/pen/YzvNvPp

**Comment by geospatialem** [comment]
Solved with https://github.com/Esri/calcite-components/pull/5725.

![verify-popover-close](https://user-images.githubusercontent.com/5023024/202747601-8aa69c1f-3ced-462c-9e7a-5270f28d0e2a.gif)
